packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:math';
+
+import '../../rendering.dart';
+
+class SliverFlexParentData extends SliverPhysicalParentData with ContainerParentDataMixin<RenderSliver> {
+  int flex = 1;
+  double extent = 0.0;
+}
+
+/// A sliver that places multiple sliver children in a linear array along the cross
+/// axis.
+/// 
+/// Since the extent of the viewport in the cross axis direction is finite,
+/// each sliver is forced to divide up the extent per specification via
+/// ParentData widgets SliverCrossAxisConstrained and SliverExpanded.
+/// 
+/// The algorithm for dividing up the cross axis extent is as follows.
+/// Every widget either has constrained cross axis extent or a flex value.
+/// First allocate space to all the widgets with constrained cross axis extent.
+/// Then divide up the remaining space proportional to the widget's flex value.
+/// Flex values can be specified via the SliverExpanded widgets, but if
+/// a sliver without SliverExpanded or SliverCrossAxisConstrained is passed,
+/// it is assumed to have a flex value of 1, which by itself simply expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverFlexParentData> {
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverFlexParentData) {
+      child.parentData = SliverFlexParentData();
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // Iterate through each child.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverFlexParentData childParentData = child.parentData! as SliverFlexParentData;
+      if (childParentData.extent == 0.0) {
```

Comment:
It may be easier to make these properties nullable and use that as a flag for using one or the other? What if the user intentionally set the extent to 0? This would override that and have it flex instead.

Reply:
Yeah I think that makes a lot of sense. I'll go with extent being nullable since by default, Slivers should try to fill up the cross axis.